Scripts/SAPSetupAssistantValidationScripts/SetupAssistantSAPValidateNetworkConnection.ps1

```diff
@@ -0,0 +1,39 @@
+<#
+.SYNOPSIS
+    Tests network connection to a specified host and port.
+.DESCRIPTION
+    This script attempts to establish a TCP connection to the specified host and port,
+    and reports the result in a JSON format.
+.PARAMETER HostName
+    The hostname or IP address to connect to.
+.PARAMETER Port
+    The port number to connect to.
+#>
+param (
+   [Parameter(Mandatory=$true)]
+   [string]$hostName,
+
+   [Parameter(Mandatory=$true)]
+   [string]$port
+)
+
+$job = Start-Job -ScriptBlock {
+    param($hostName, $port)
+    Test-NetConnection -ComputerName $hostName -Port $port
+} -ArgumentList $hostName, $port
+
+if (Wait-Job -Job $job -Timeout 5) {
```

Comment:
5 May or may not be a sufficient time. You could allow this to be passed in so it's configurable, but with a default. And consider making the default longer.

Reply:
Increased to 10 sec , we want to limit the parameters sent by users . So updated the default itself